src/commands/index.rs

```diff
@@ -170,9 +177,9 @@ fn scan_and_index(ctx: &Context, schema: &Schema, cursor: u64) -> Result<u64, Re
                             .get_value::<RedisJSON>(&REDIS_JSON_TYPE)
                             .and_then(|doc| {
                                 if let Some(data) = doc {
-                                    if let Some(index) = &data.index {
-                                        if schema.name == *index {
-                                            add_document(key, index, data)?;
+                                    if let Some(index) = &data.value_index {
+                                        if schema.name == index.index {
+                                            add_document(key, &index.index, data)?;
```

Comment:
Please rename the variable inside the pattern match for consistency:

```
if let Some(value_index) = &data.value_index {
    if schema.name == value_index.index {
        add_document(key, &value_index.index, data)?;
```

Reply:
:+1: